src/main/java/seedu/address/logic/parser/FindTransactionCommandParser.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.Arrays;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.FindTransactionCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.TransactionContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindTransactionCommand object
+ */
+public class FindTransactionCommandParser implements Parser<FindTransactionCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindTransactionCommand
+     * and returns a FindTransactionCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public FindTransactionCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindTransactionCommand.MESSAGE_USAGE));
+        }
+
+        // first item is index, the rest are keywords
+        String[] transactionKeywords = trimmedArgs.split("\\s+");
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(transactionKeywords[0]);
+            // remove the first item (index)
+            transactionKeywords = Arrays.copyOfRange(transactionKeywords, 1, transactionKeywords.length);
+        } catch (ParseException e) {
+            // if index is not provided, default to first index
+            index = Index.fromOneBased(1);
+        }
```

Comment:
I think you should use `model.getIsViewTransaction()` to check if the user is in the transaction view. The try-catch block should still be reserved for if the user types the wrong number of inputs. For example, if user is in person list view but types `find {keyword}`

Reply:
I want to do that too but we are not passing `model` as argument into the parser 😢

Reply:
hmm should we just make them specify an index then? Because i think if we default to 1 it is buggy

Reply:
So in the transaction view they need to specify index 1?

Reply:
hmm ig

Reply:
I think we just tell the users not to use any command that needs to specify the person index in the transaction view.

Reply:
Becuase we also cannot use `addt`

Reply:
Updated in [#4dbd4b8](https://github.com/AY2425S1-CS2103T-F13-4/tp/pull/66/commits/4dbd4b8a5d83556d4bad2a73e784ce08b0507f6a)

Reply:
ok it looks good now